ee/tabby-webserver/src/handler.rs

```diff
@@ -1,51 +1,107 @@
 use std::sync::Arc;
 
+use async_trait::async_trait;
 use axum::{
     extract::{Path, State},
     http::Request,
     middleware::{from_fn_with_state, Next},
     response::{IntoResponse, Response},
     routing, Extension, Json, Router,
 };
+use cached::{CachedAsync, TimedCache};
 use hyper::{header::CONTENT_TYPE, Body, StatusCode};
 use juniper::ID;
 use juniper_axum::{extract::AuthBearer, graphiql, graphql, playground};
-use tabby_common::api::{
-    code::CodeSearch,
-    event::{ComposedLogger, EventLogger},
-    server_setting::ServerSetting,
+use tabby_common::{
+    api::{
+        code::CodeSearch,
+        event::{ComposedLogger, EventLogger},
+        server_setting::ServerSetting,
+    },
+    config::{RepositoryAccess, RepositoryConfig},
 };
 use tabby_db::DbConn;
+use tokio::sync::Mutex;
 use tracing::{error, warn};
 
 use crate::{
     cron, hub, integrations, oauth,
     path::db_file,
-    repositories::{self},
-    schema::{auth::AuthenticationService, create_schema, Schema, ServiceLocator},
+    repositories,
+    schema::{
+        auth::AuthenticationService, create_schema, repository::RepositoryService, Schema,
+        ServiceLocator,
+    },
     service::{create_service_locator, event_logger::create_event_logger},
     ui,
 };
 
+struct ServiceRepositoryAccess {
```

Comment:
```suggestion
struct RepositoryAccessImpl {
```

Reply:
Didn't want to call it this because we have other implementations of `RepositoryAccess`, I think this is more clear since it is the implementation of `RepositoryAccess` which is based on the service as opposed to config.

Reply:
Service is too generic - since this srruct is private to the file, `Impl` should be a sufficient good default.